modules/assim.sequential/R/sda.enkf_MultiSite.R

```diff
@@ -142,338 +151,464 @@ sda.enkf.multisite <- function(settings, obs.mean, obs.cov, Q = NULL, restart=F,
+      #------------------------------------------- Reading the output
+      if (control$debug) browser()
+      #--- Reading just the first run when we have all years and for VIS
+      
+      if (t==1 & control$FF){
+        readsFF <- out.configs %>%
+          purrr::map(function(configs) {
+            
+            obs.times%>%
+              purrr::map(function(stop.date){
+                X_tmp <- vector("list", 2)
+                
+                for (i in seq_len(nens)) {
+                  X_tmp[[i]] <- do.call( my.read_restart,
+                                         args = list(
+                                           outdir = outdir,
+                                           runid = configs$runs$id[i] %>% as.character(),
+                                           stop.time =  stop.date,
+                                           settings = settings,
+                                           var.names = var.names,
+                                           params = new.params[[i]]
+                                         )
+                  )
+                  
+                }
+                
+                return(X_tmp %>% map_df(~ .x[['X']] %>% t %>%
+                                          as.data.frame %>% 
+                                          mutate(Date=stop.date,
+                                                 Site=(configs$runs$id[i] %>%strsplit('-') %>% unlist())[3] ))
+                )
+              }) %>% setNames(obs.times)
+            
+          })
+        
       }
+      #------------- Reading - every iteration and for SDA
+      reads <- out.configs %>%
+        `class<-`(c("list")) %>%
+        furrr::future_map(function(configs) {
+          # Loading the model package - this is required bc of the furrr
+          library(paste0("PEcAn.",model), character.only = TRUE)
+          
+          X_tmp <- vector("list", 2)
+
+          for (i in seq_len(nens)) {
+            X_tmp[[i]] <- do.call( my.read_restart,
+                                   args = list(
+                                     outdir = outdir,
+                                     runid = configs$runs$id[i] %>% as.character(),
+                                     stop.time = obs.times[t],
+                                     settings = settings,
+                                     var.names = var.names,
+                                     params = new.params[[i]]
+                                   )
+            )
+            
+          }
+          return(X_tmp)
+        })
+      
+      if (control$debug) browser()
+      #let's read the parameters of each site/ens
+      params.list <- reads %>% map(~.x %>% map("params"))
+      # Now let's read the state variables of site/ens
+      X <- reads %>% map(~.x %>% map_df(~.x[["X"]] %>% t %>% as.data.frame))
+      
+      
+      #replacing crazy outliers before it's too late
+      X <- outlier.detector.boxplot(X)
```

Comment:
This step seems dangerous, especially as it's going to be occurring silently without the end user knowing it's here. If this has value, it should be controlled explicitly by a flag, and that flag should default to OFF.